### DISCUSSION TITLE
Is it possible to toggle diagnostics on LSP?

### DISCUSSION THREAD

**Discussion by JapandiTech** [helix-editor/helix#5402] [Q&A]
Hi, I would like to disable diagnostics from my LSP. As of now I've only found diagnostic-severity where I can minimally set to be shown errors. But I would like to be shown nothing at all. Is there something for this yet or do I need to make a pull request?

**Comment by JapandiTech** [comment]
No one who knows if it's possible to disable diagnostics?

**Comment by David-Else** [comment]
I don't think so, but I see there is a pull request relating to toggling waiting for review: https://github.com/helix-editor/helix/pull/4085/files . I don't know if that could be used for your request somehow.

Maybe when the new virtual text feature is merged then you could toggle virtual text on and off in the future?

**Comment by nathan-at-least** [comment]
+1 to this.

I may be a dinosaur, but I find diagnostics distracting, especially when I'm in the midst of change which I _know_ introduce temporary failures (for example, because I need to edit multiple parts of code in-sync such as changing both an interface and call sites). OTOH, I _do_ want diagnostics when I believe I've completed that sequence of edits and want to check my work.

So my ideal would be a hot key to toggle diagnostic display.

Meanwhile, I want syntax, semantic-navigation (ie "go to definition"), auto-formatting so I don't want to disable LSP altogether.

**Reply by masonmcelvain** [reply]
My present workaround is `:lsp-stop`, make change that introduces temporary failures, then `:lsp-restart`. Not ideal but it's something.
